SAMtags.tex

```diff
@@ -317,11 +320,30 @@ \subsection{Barcodes}
 \subsection{Original data}
 
 \begin{description}
+\item[OA:Z:\tagregex{{\tt (}\sf{RNAME}{\tt,\,}\sf{POS}{\tt,\,}\emph{strand}{\tt,\,}\sf{CIGAR}{\tt,\,}\sf{MAPQ}{\tt,\,}\sf{NM}{\tt; )}+}]
+The tag encodes the original alignment information of the record prior to realignment or unalignment by subsequent tool. 
+Here \emph{Strand} is either `{\tt +}' or `{\tt -}', indicating whether the read has been reverse-complemented, which corresponds to FLAG bit 0x10.
+That is, \emph{Strand} is `+' if the bit is unset and `-' if the bit is set.
+The resulting string is the result of joining the 6 values with commas `,' (no additional spaces).
+If  {\tt NM} is unknown it may be omitted, but the comma delimiter must be retained so that the string can be easily parsed into the 6 subfields.
+In the presence of an existing {\tt OA} tag, a subsequent tool may append another set of original alignment information after the semicolon,
+adding to---rather than replacing---the existing {\tt OA} information.
+
+Note 1: While commas are legal characters in sequence names, it is forbidden to have an {\tt OA} tag with such sequences.
```

Comment:
What's about adding a recomendation or enforce an invalid character in the sequence name to encode the `,`? I prefer the second option, as it enforce some guidelines for encode/decode the tag - otherwise, it will be implementation dependent and will make incompatible OA tags in the wild. For example, a tool for regenerate the alignments of a BAM file where some sequence name has commas (`my,sequence`), might generate a record mapped to an unknown RNAME (`my_sequence`, if that was the workaround).

Reply:
If I recall there are plans to change the spec to make comma illegal ref names, given they're not currently used in any noticable amounts (see https://github.com/samtools/hts-specs/issues/291).  Id' rather go down this route, even if it's an adjustment to the spec version numbers (1.6 onwards means commas and some other chars are forbidden), than putting something in place which further promotes the idea that we can put anything we like in a name.

Reply:
Fair enough - but that means that OA won't be accepted after #291 is addressed? Otherwise, what will happen with v1.5 files after this gets in, if there are a file in the wild with commas in the RNAME?